robot/impl/robot_reconfigure_test.go

```diff
@@ -3698,6 +3699,82 @@ func TestResourceConstructTimeout(t *testing.T) {
 	rr.reconfigureWorkers.Wait()
 }
 
+func TestResourceConstructCtxCancel(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+
+	contructCount := 0
+	var wg sync.WaitGroup
+
+	mockAPI := resource.APINamespaceRDK.WithComponentType("mock")
+	modelName1 := utils.RandomAlphaString(5)
```

Comment:
[minor] does this need to be random or can we just make it an obviously fake name? I think the latter is easier to debug should this test fail.

```suggestion
	modelName1 := "fakeModel"
```

Reply:
random is slightly better so as to not run into issues registering multiple fakes across different tests (shouldn't be as bad nowadays since we deregister, but tests running in parallel might still run into issues

Reply:
gotcha - in that case could we use a `uuid` string to prevent a very unfortunately collision across parallel tests?

Reply:
I think RandomAlphaString is suitably random, can spin it out to a separate ticket to update all references if we really want